core/rawdb/accessors_chain.go

```diff
@@ -1166,7 +1182,15 @@ func ReadEtxSet(db ethdb.Reader, hash common.Hash, number uint64) types.EtxSet {
 	}
 	etxSet := make(types.EtxSet)
 	for _, entry := range entries {
-		etxSet[entry.EtxHash] = types.EtxSetEntry{Height: entry.EtxHeight, ETX: entry.Etx}
+		etxRlp := ReadEtxRLP(db, entry.EtxHash)
+		if etxRlp == nil {
+			return nil
+		}
+		var etx types.Transaction
+		if err := rlp.Decode(bytes.NewReader(etxRlp), etx); err != nil {
+			return nil
```

Comment:
Is it feasible to make this function return the error here instead of nil?
At least I think we should log it.

Reply:
I'll log it, but I can't change the interface. The function is infallible

Reply:
fixed